include/maths/CTimeSeriesDecompositionDetail.h

```diff
@@ -181,8 +204,111 @@ class MATHS_EXPORT CTimeSeriesDecompositionDetail : private CTimeSeriesDecomposi
         THandlerRefVec m_Handlers;
     };
 
-    //! \brief Scans through increasingly low frequencies looking for custom
-    //! diurnal and any other large amplitude seasonal components.
+    //! \brief Checks for sudden change or change events.
+    class MATHS_EXPORT CChangePointTest : public CHandler {
```

Comment:
For definite this needs restructuring. I want to create subdirectories in `maths` as you suggest. Then this can all move into a details subdirectory, for time series modelling, and be split over several files. However, I don't want to make this change in this PR, which is already large. It also needs us to do some work to get the build system to accept the new directory arrangement. So whilst I agree with this comment, I would propose making the change in a separate PR. WDYT?

Reply:
+1 Let's have it in a separate PR